### ISSUE TITLE
A few curious questions about this implementation

### ISSUE THREAD

**Issue by kierkegaard13** [rosinality/stylegan2-pytorch#148]
Hello,

I've been looking at a few of the pytorch implementations and the stylegan papers for references to create my own implementation (for learning purposes), and I was curious if you knew the difference between the pytorch bilinear sampling and your implementation here.

I also noticed your FusedLeakyReLU and ScaledLeakyReLU functions, and I was wondering where you got those from. Were those mentioned somewhere in the paper or in the official implementation? The ScaledLeakyReLU function definitely seemed to fix some issues I was having during training though I'm not entirely sure why other than that this architecture seems to be heavily dependent on getting the weight initializations and normalizations correct.

Thanks.

**Comment by rosinality** [comment]
For upsampling the size of blur kernel is larger (4) than that will be used bilinear upsampling (3), and behavior is different on the boundary of feature map. As 0.5x downsampling for bilinear downsampling is same with average pooling, so it is also different.

Fused activations is mentioned in the paper and you can find implementation in the official implementation. As stylegan2 does not use feature map normalizations like instance norm, it will be important to use appropriate initializations, scaling and weight normalizations.

**Comment by kierkegaard13** [comment]
Awesome. Thanks for the reply.
